File: NavigationDemo/app/src/main/java/com/finham/navigationdemo/MyViewModel.java
package com.finham.navigationdemo;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

/**
 * User: Fin
 * Date: 2020/2/1
 * Time: 19:14
 */
public class MyViewModel extends ViewModel {
    private MutableLiveData<Integer> number;

    public MutableLiveData<Integer> getNumber() {
        if (number == null) {
            number = new MutableLiveData<>();
            number.setValue(0);
        }
        return number;
    }

    public void add(int x) {
        number.setValue(getNumber().getValue()+x);
        if(number.getValue()<0){
            number.setValue(0);
        }
    }
}

File: KotlinDemo/app/src/main/java/com/finham/kotlindemo/MyViewModel.kt
package com.finham.kotlindemo

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * User: Fin
 * Date: 2020/2/29
 * Time: 20:45
 */
class MyViewModel : ViewModel() {
    private val _number: MutableLiveData<Int> by lazy { MutableLiveData<Int>().also { it.value = 0 } } //不需要改，只需要赋值一次初始化即可，所以设为val
    val number: LiveData<Int>
    get() = _number
    //val number1=MutableLiveData(0)

    fun modifyNumber(modifiedAmount: Int) {
        _number.value = _number.value?.plus(modifiedAmount)
    }
}